### ISSUE TITLE
[Bug]: Slider, onRelease and onBlur

### ISSUE THREAD

**Issue by stanley-regnard** [carbon-design-system/carbon#12187]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.13.0

### React version

18.2.0

### Description

When typing in the TextInput attached to a Slider, each character typed will trigger the onRelease function.
Exiting the same TextInput will result in a error logged in the browser console:
```
Uncaught TypeError: _this.props.onBlur is not a function
    at @carbon_react.js?v=0c3423c8:65038:19
    at HTMLUnknownElement.callCallback2 (chunk-HSVJWAOT.js?v=7bb9af7f:3793:22)
    at Object.invokeGuardedCallbackDev (chunk-HSVJWAOT.js?v=7bb9af7f:3818:24)
    at invokeGuardedCallback (chunk-HSVJWAOT.js?v=7bb9af7f:3852:39)
    at invokeGuardedCallbackAndCatchFirstError (chunk-HSVJWAOT.js?v=7bb9af7f:3855:33)
    at executeDispatch (chunk-HSVJWAOT.js?v=7bb9af7f:6484:11)
    at processDispatchQueueItemsInOrder (chunk-HSVJWAOT.js?v=7bb9af7f:6504:15)
    at processDispatchQueue (chunk-HSVJWAOT.js?v=7bb9af7f:6513:13)
    at dispatchEventsForPlugins (chunk-HSVJWAOT.js?v=7bb9af7f:6521:11)
    at chunk-HSVJWAOT.js?v=7bb9af7f:6645:20
```

### Reproduction/example

https://stackblitz.com/edit/github-h8unbj?file=src%2FApp.jsx

### Steps to reproduce

If you use the Slider, onChange and onRelease values should update as expected.
While typing in the TextInput, you would notice that both onChange and onRelease values will be edited live.
Exiting the TextInput should trigger an error in the console.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by jnm2377** [comment]
Can confirm I am also seeing this error in our storybook.

**Comment by stanley-regnard** [comment]
It looks like https://github.com/carbon-design-system/carbon/pull/12207 should fix the error with onBlur
However that probably shouldn't close this issue, there's still the (more problematic) bug with onRelease being triggered by the TextInput

**Comment by abbeyhrt** [comment]
@stanley-regnard Agreed, there are kind of two bugs here, what's going on with `onBlur` that is resolved by #12207 and `onRelease` being called when it shouldn't. We'll adjust the other PR to not close this one
